advanced-usage/index.md

```diff
@@ -0,0 +1,372 @@
+- Unprotected endpoints can be removed or disabled:
+{% end %}
+{{ easy_image(src="endpoints-disable-delete", width=400, class="pirate") }}
+{% pirate() %}
+- Modifying an endpoint is not possible - a new one must be created instead
+   - e.g. some users may wish to set up a UDP endpoint for connecting to with
+   Pymavlink from the surface:
+{% end %}
+{{ easy_image(src="endpoints-pymavlink-example", width=400, alt_extra="Example", class="pirate") }}
+
+## Tools
+
+### System Information
+The system information page provides useful information about the processes,
+network configuration, and computer system BlueOS is running on. It can be
+useful for troubleshooting, and finding if a particular program is using
+excessive resources.
+{{ easy_image(src="system-info-processes", width=600) }}
+{{ easy_image(src="system-info-monitor", width=600) }}
+{{ easy_image(src="system-info-network", width=600) }}
+{{ easy_image(src="system-info-about", width=600) }}
+
+### Network Test
+The Network Test page measures real-time latency between BlueOS and the surface
+computer, and allows checking the upload and download speeds between them.
+
+{{ easy_image(src="network-test", width=600) }}
+
+### Version Chooser
+The Version Chooser is a major component in the robust backbone of BlueOS. It
+runs independently from the main interface, and is monitored such that if it
+somehow fails a backup version will be run in its place.
+
+{{ simple_pirate_image(src="version-chooser", width=600) }}
+
+- The simplified interface provides an easy way to update to the latest version
+that is [as stable or more stable](../overview/#release-types) than the currently installed version
+{% pirate() %}
+- The full interface supports easily changing forwards _and backwards_ between
+versions
+   - Previously-installed versions are kept locally on the device, unless
+   manually deleted, which provides an easy route for roll-backs to undesired
+   changes (e.g. during development)
+- Allows loading remote versions (including from custom docker-hub repositories)
+- Allows manually uploading docker images from the surface computer
+- If an undetected failure somehow occurs in BlueOS (or if a broken version gets
+installed) it's possible to easily roll back to a working version from
+   - on the device
+   - manual upload, or
+   - downloaded from the internet
+- If necessary, the underlying service can be accessed directly at TCP port `8081`
+   - e.g. http://blueos.local:8081
+{% end %}
+
+{% pirate() %}
+### Available Services
+
+The Available Services page provides developer access to the underlying http
+server interfaces of the services upon which BlueOS is based. Each service is
+listed with 
+- the port it is served at
+- a meaningful name
+- a web-page link using the active 
+[network configuration](../getting-started/#interface-access), and where relevant
+- its API documentation (in a live-testable form)
+- the current API version
+{% end %}
```

Comment:
Any service that provides an http server with a [title tag](https://www.w3schools.com/tags/tag_title.asp) will be displayed, documentation can also be parsed if follows swagger/open api spec and available in `/docs` or `/v1.0/ui/`